packages/graphql/src/mutations/uniswap/swapToToken.js

```diff
@@ -3,13 +3,23 @@ import contracts from '../../contracts'
 
 export async function swapToTokenTx(tokenValue) {
   const exchange = contracts.daiExchangeExec
+
+  // The uniswap contract requires to specify a deadline which is a timestamp
+  // in second after which the transaction can no longer be executed.
+  // See method tokenToExchangeSwapOutput here:
+  //   https://github.com/Uniswap/contracts-vyper/blob/master/contracts/uniswap_exchange.vy
+  //
+  // Under normal condition it should not take more than a couple minutes
+  // for the transaction to get mined. But in case the network is extremely
+  // congested, we set the deadline to a conservative value of 1 hour from now.
+  //
+  // We use max of (localTimestamp, block.timestamp) to calculate the current time because:
+  //  - The local clock could be skewed
+  //  - When running in test/dev environment blocks may not be mined for some time.
   const blockNumber = await contracts.web3.eth.getBlockNumber()
   const block = await contracts.web3.eth.getBlock(blockNumber)
-
-  // If we're running on a private blockchain that hasn't mined a block
-  // recently, use a more recent timestamp
-  const now = Math.round(+new Date() / 1000)
-  const deadline = (block.timestamp < now - 60 ? now : block.timestamp) + 300
+  const localTimestamp = Math.round(+new Date() / 1000)
```

Comment:
What does the `+` mean in this context?

Reply:
Fancy JS notation. It it the unary operator. It converts the date to a number.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus

Reply:
That's pretty cool, thanks.  Will keep it in mind.